x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRCS1DeprecationIT.java

```diff
@@ -0,0 +1,138 @@
+            var putRoleRequest = new Request("PUT", "/_security/role/" + REMOTE_SEARCH_ROLE);
+            putRoleRequest.setJsonEntity("""
+                {
+                  "indices": [
+                    {
+                      "names": ["local_index"],
+                      "privileges": ["read"]
+                    }
+                  ]
+                }""");
+            assertOK(adminClient().performRequest(putRoleRequest));
+
+            // Query cluster -> create user and assign role
+            var putUserRequest = new Request("PUT", "/_security/user/" + REMOTE_SEARCH_USER);
+            putUserRequest.setJsonEntity("""
+                {
+                  "password": "x-pack-test-password",
+                  "roles" : ["remote_search"]
+                }""");
+            assertOK(adminClient().performRequest(putUserRequest));
+
+            // Query cluster -> create test index
+            var indexDocRequest = new Request("POST", "/local_index/_doc?refresh=true");
+            indexDocRequest.setJsonEntity("{\"local_foo\": \"local_bar\"}");
+            assertOK(client().performRequest(indexDocRequest));
+
+            // Fulfilling cluster -> create test indices
+            Request bulkRequest = new Request("POST", "/_bulk?refresh=true");
+            bulkRequest.setJsonEntity(Strings.format("""
+                { "index": { "_index": "index1" } }
+                { "foo": "bar" }
+                { "index": { "_index": "secretindex" } }
+                { "bar": "foo" }
+                """));
+            assertOK(performRequestAgainstFulfillingCluster(bulkRequest));
+
+            // Fulfilling cluster -> add role for remote search user
+            var putRoleOnRemoteClusterRequest = new Request("PUT", "/_security/role/" + REMOTE_SEARCH_ROLE);
+            putRoleOnRemoteClusterRequest.setJsonEntity("""
+                {
+                  "indices": [
+                    {
+                      "names": ["index*"],
+                      "privileges": ["read", "read_cross_cluster"]
+                    }
+                  ]
+                }""");
+            assertOK(performRequestAgainstFulfillingCluster(putRoleOnRemoteClusterRequest));
+        }
+        {
+            // perform a simple search request, so we can ensure the remote cluster is connected
+            final Request searchRequest = new Request(
+                "GET",
+                String.format(
+                    Locale.ROOT,
+                    "/%s:index1/_search?ccs_minimize_roundtrips=%s",
+                    randomFrom(REMOTE_CLUSTER_ALIAS, "*", "my_remote_*"),
+                    randomBoolean()
+                )
+            );
+            assertOK(performRequestWithRemoteSearchUser(searchRequest));
+        }
+        {
+            // verify that the deprecation warning is logged
+            try (InputStream log = queryCluster.getNodeLog(0, LogType.DEPRECATION)) {
+                Streams.readAllLines(
```

Comment:
Can be a follow up but might also check (for any of the existing suites) that we are _not_ getting the deprecation log for RCS 2.0

Reply:
Good call-out, I've pushed https://github.com/elastic/elasticsearch/pull/120806/commits/40f3773fe5c0353937bd16ce3a6513f1ace6eab4